hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/ReadableFreezeUpgradeActions.java

```diff
@@ -235,6 +233,13 @@ private void extractAndReplaceArtifacts(
         }
     }
 
+    private long getNextNodeID(@NonNull List<ActiveNode> nodes) {
+        requireNonNull(nodes);
+        final long maxNodeId =
+                nodes.stream().mapToLong(a -> a.node.nodeId()).max().orElse(-1L);
+        return maxNodeId + 1;
+    }
+
```

Comment:
Can we move this also to `AddressBookHelper` ? Can we re-use one version of this method?

Reply:
The reason I created AddressBookHelper initially is tried to re-use getNextNodeID(@NonNull final ReadableNodeStore nodeStore) here. But I kept on getting failed immutable exception during DabEnabledUpgradeTest by using it, I was going to check with you and @tinker-michaelj later. This small function get me to pass that error for now.

Reply:
Since this is called in ExecutorService thread, we can't use ReadableStore directly.